memphis/memphis.go

```diff
@@ -28,17 +29,66 @@ type MemphisOutput struct {
 	FailedMessages []MemphisMsgWithError `json:"failed_messages"`
 }
 
-// EventHandlerFunction gets the message payload as []byte, message headers as map[string]string and inputs as map[string]string and should return the modified payload and headers.
+// HandlerType functions get the message payload as []byte (or any), message headers as map[string]string and inputs as map[string]string and should return the modified payload and headers.
 // error should be returned if the message should be considered failed and go into the dead-letter station.
 // if all returned values are nil the message will be filtered out of the station.
-type EventHandlerFunction func([]byte, map[string]string, map[string]string) ([]byte, map[string]string, error)
+type HandlerType func(any, map[string]string, map[string]string) (any, map[string]string, error)
+
+type PayloadOption func(*PayloadOptions) error
+
+type PayloadOptions struct {
+	Handler     HandlerType
+	UserObject  any
+	PayloadType PayloadTypes
+}
+
+type PayloadTypes int
+
+const (
+	BYTES PayloadTypes = iota + 1 
+	JSON 
+)
+
+func PayloadAsJSON(schema any) PayloadOption {
+	return func(payloadOptions *PayloadOptions) error {
+		payloadOptions.UserObject = schema
+		payloadOptions.PayloadType = JSON
+		return nil
+	}
+}
+
+func UnmarshalIntoStruct(data []byte, userStruct any) error {
+	// Unmarshal JSON data into the struct
+	err := json.Unmarshal(data, userStruct)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
 
 // This function creates a Memphis function and processes events with the passed-in eventHandler function.
-// eventHandlerFunction gets the message payload as []byte, message headers as map[string]string and inputs as map[string]string and should return the modified payload and headers.
+// eventHandler gets the message payload as []byte or as the user specified type,
+// message headers as map[string]string and inputs as map[string]string and should return the modified payload and headers.
+// The modified payload type will either be the user type, or []byte depending on user requirements.
 // error should be returned if the message should be considered failed and go into the dead-letter station.
 // if all returned values are nil the message will be filtered out from the station.
-func CreateFunction(eventHandler EventHandlerFunction) {
+func CreateFunction(eventHandler HandlerType, options ...PayloadOption) {
 	LambdaHandler := func(ctx context.Context, event *MemphisEvent) (*MemphisOutput, error) {
+		params := PayloadOptions{
+			Handler:    eventHandler,
+			UserObject: nil,
```

Comment:
Let's name it UserStruct instead, in Go using 'object' wouldn't be quite right

Reply:
I renamed it